src/plugin.ts

```diff
@@ -86,12 +86,15 @@ task<CompileJobArgs>(TASK_COMPILE_SOLIDITY_COMPILE_JOB, async (args, hre, superC
 });
 
 async function getExposedJob(hre: HardhatRuntimeEnvironment, compilationJob: CompilationJob, output: CompilerOutput): Promise<CompilationJob> {
+  const path = await import('path');
+  const { isMatch } = await import('micromatch');
   const { getExposed } = await import('./core');
 
-  const inputFiles = Object.fromEntries(compilationJob.getResolvedFiles().map(rf => [rf.sourceName, rf.absolutePath]));
+  const sourcesDir = path.relative(hre.config.paths.root, hre.config.paths.sources);
 
-  const include = await getMatcher(hre.config);
-  const exposed = getExposed(output, include, hre.config);
+  const include = (sourceName: string) => sourceName.startsWith(sourcesDir) && hre.config.exposed.include.some(p => isMatch(path.relative(sourcesDir, sourceName), p));
```

Comment:
I'm not sure if the `sourceName.startsWith(sourcesDir)` is really necessary. The `hre.config.exposed.include`is likelly to only allow these path anyway.

Reply:
```suggestion
  const include = (sourceName: string) => hre.config.exposed.include.some(p => isMatch(path.relative(sourcesDir, sourceName), p));
```